src/Catalyst.Core.Modules.Dfs/UnixFs/ChunkedStream.cs

```diff
@@ -0,0 +1,179 @@
+        private long fileSize;
+
+        /// <summary>
+        ///   Creates a new instance of the <see cref="ChunkedStream"/> class with
+        ///   the specified <see cref="IBlockApi"/> and <see cref="DagNode"/>.
+        /// </summary>
+        /// <param name="blockService"></param>
+        /// <param name="keyChain"></param>
+        /// <param name="dag"></param>
+        internal ChunkedStream(IBlockApi blockService, IKeyApi keyChain, IDagNode dag)
+        {
+            BlockService = blockService;
+            KeyChain = keyChain;
+            var links = dag.Links.ToArray();
+            var dm = Serializer.Deserialize<DataMessage>(dag.DataStream);
+            if (dm.FileSize != null) fileSize = (long) dm.FileSize;
+            ulong position = 0;
+            for (var i = 0; i < dm.BlockSizes.Length; ++i)
+            {
+                blocks.Add(new BlockInfo
+                {
+                    Id = links[i].Id,
+                    Position = (long) position
+                });
+                position += dm.BlockSizes[i];
+            }
+        }
+
+        private IBlockApi BlockService { get; }
+        private IKeyApi KeyChain { get; }
+
+        /// <inheritdoc />
+        public override long Length => fileSize;
+
+        /// <inheritdoc />
+        public override void SetLength(long value) { throw new NotSupportedException(); }
+
+        /// <inheritdoc />
+        public override bool CanRead => true;
+
+        /// <inheritdoc />
+        public override bool CanSeek => true;
+
+        /// <inheritdoc />
+        public override bool CanWrite => false;
+
+        /// <inheritdoc />
+        public override void Flush() { }
+
+        /// <inheritdoc />
+        public override long Position { get; set; }
+
+        /// <inheritdoc />
+        public override long Seek(long offset, SeekOrigin origin)
+        {
+            switch (origin)
+            {
+                case SeekOrigin.Begin:
+                    Position = offset;
+                    break;
+                case SeekOrigin.Current:
+                    Position += offset;
+                    break;
+                case SeekOrigin.End:
+                    Position = Length - offset;
+                    break;
+                default:
+                    throw new ArgumentOutOfRangeException(nameof(origin), origin, null);
+            }
+
+            return Position;
+        }
+
+        /// <inheritdoc />
+        public override void Write(byte[] buffer, int offset, int count) { throw new NotSupportedException(); }
+
+        /// <inheritdoc />
+        public override int Read(byte[] buffer, int offset, int count)
+        {
+            return ReadAsync(buffer, offset, count).ConfigureAwait(false).GetAwaiter().GetResult();
+        }
+
+        /// <inheritdoc />
+        public override async Task<int> ReadAsync(byte[] buffer, int offset, int count, CancellationToken cancel)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Rename parameter 'cancel' to 'cancellationToken' to match the base class declaration.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)